# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../musl/pkg/strace/strace-linux-ptrace-header.patch
# Copyright (C) 2014 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Original patch was taken from sabotage linux:

https://github.com/sabotage-linux/sabotage/blob/master/KEEP/strace-kernelhdr_3.12.6.patch

The second part which includes sys/sysinfo.h in the original patch needed
to be removed to be able to compile strace with OpenSDE's current kernel
headers of linux 3.10.29.

--- strace-4.8.org/process.c
+++ strace-4.8/process.c
@@ -55,19 +55,6 @@
 # endif
 #endif
 
-#ifdef HAVE_LINUX_PTRACE_H
-# undef PTRACE_SYSCALL
-# ifdef HAVE_STRUCT_IA64_FPREG
-#  define ia64_fpreg XXX_ia64_fpreg
-# endif
-# ifdef HAVE_STRUCT_PT_ALL_USER_REGS
-#  define pt_all_user_regs XXX_pt_all_user_regs
-# endif
-# include <linux/ptrace.h>
-# undef ia64_fpreg
-# undef pt_all_user_regs
-#endif
-
 #if defined(SPARC64)
 # define r_pc r_tpc
 # undef PTRACE_GETREGS
